src/main/java/frc/robot/commands/OrbitalTarget.java

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import com.pathplanner.lib.util.PIDConstants;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.classes.Util;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.chassis.Chassis;
+import java.util.function.DoubleSupplier;
+import org.littletonrobotics.junction.Logger;
+
+public class OrbitalTarget extends Command {
+
+  private final Chassis chassis;
+
+  private final DoubleSupplier xSupplier;
+  private final DoubleSupplier ySupplier;
+  private final DoubleSupplier lTriggerSupplier;
+  private final DoubleSupplier rTriggerSupplier;
+
+  private final Translation2d targetPose;
+
+  private final PIDController xController;
+  private final PIDController yController;
+  private final PIDController rotController;
+
+  private PIDConstants translationPID;
+  private PIDConstants rotationPID;
+  private double maxSpeed;
+
+  // Target pose in field space for the robot to move to
+  private double xTarget;
+  private double yTarget;
+  private double rotTarget;
+
+  // Essentially the polar coordinates of the robot relative to the target
+  private double TRAngle; // Target-Robot angle
```

Comment:
If you need a comment to tell you what the variable means, you should change the name of the variable.